userland/src/parameters.cpp

```diff
@@ -0,0 +1,74 @@
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+template <typename ParameterTypeT>
+ParameterTypeT
+wait_for_future(rclcpp::node::Node::SharedPtr & node, std::shared_future<ParameterTypeT> & future)
+{
+  std::future_status status;
+  do {
+    rclcpp::spin_some(node);
+    status = future.wait_for(std::chrono::milliseconds(100));
+  } while (status != std::future_status::ready && rclcpp::ok());
+
+  if (std::future_status::ready == status) {
+    return future.get();
+  }
+
+  throw std::runtime_error("Invalid future status");
+}
```

Comment:
> I'd rather add this to the synchronous version of the client

Can you explain in more detail what you mean by adding it to the synchronous version? As a function or a method?

The users using the async API also need this, consider:

```c++
auto f = node->async_has_parameter(...);
// User does some work in the mean time
bool found = wait_for_future<bool>(node, f);
```

I think this function provides utility that is common enough that it should be included with the standard async API.